### PR TITLE
Implement _taskDoneListener using PlanCompletionListener

### PR DESCRIPTION
**Pull request by angxu**
Implement _taskDoneListener using PlanCompletionListener to take sideEffect task into account.

### CODE REVIEW

File: src/com/linkedin/parseq/Engine.java [hunk 1]
```diff
@@ -99,7 +99,10 @@
     _loggerFactory = loggerFactory;
     _properties = properties;
     _planDeactivationListener = planActivityListener;
-    _planCompletionListener = planCompletionListener;
+    _planCompletionListener = planContext -> {
+      planCompletionListener.onPlanCompleted(planContext);
```
Comment (jodzga) [inline src/com/linkedin/parseq/Engine.java]: Wrap customer's code in try-catch to always call _taskDoneListener.

File: test/com/linkedin/parseq/TestEngine.java [hunk 2]
```diff
@@ -148,6 +148,33 @@ public void testShutdownWithRunningAndSuccessorTask() throws InterruptedExceptio
   }
 
   @Test
+  public void testShutdownWithSideEffectTask() throws InterruptedException {
+    final CountDownLatch finishLatch = new CountDownLatch(1);
+    final String mainValue = "main task executed";
+    final String sideEffectValue = "side-effect task executed";
+
+    Task<String> sideEffect = Task.async(context -> {
+      finishLatch.await();
```
Comment (jodzga) [inline test/com/linkedin/parseq/TestEngine.java]: Could you also add a test that is exactly the same except that it does not block a parseq thread but instead returns uncompleted promise?

Comment (angxu) [inline test/com/linkedin/parseq/TestEngine.java]: Done.

File: src/com/linkedin/parseq/Engine.java (hunk 1 above)
Comment (angxu) [inline src/com/linkedin/parseq/Engine.java]: Done.

**Comment by jodzga** [comment]
Ship it!

**Pull request merged**